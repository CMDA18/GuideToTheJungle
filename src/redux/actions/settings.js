// @flow

// Flow Types
export type TestAction = {
  type: string,
  value: string
}
export type PreloadAction = {
  type: string,
  value: string
}

// Action type constants
export const testActionType: string = 'TEST'
export const preloadActionType: string = 'PRELOAD'

// Action creators
export const createTestAction = (value: string): TestAction => ({
  type: testActionType,
  value
})

export const createPreloadAction = (value: string): PreloadAction => ({
  type: preloadActionType,
  value
})
