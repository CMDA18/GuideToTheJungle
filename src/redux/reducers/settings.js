// @flow
import { preloadActionType, testActionType, type TestAction, type PreloadAction } from '../actions/settings'

// Flow Types
export type State = {
  test?: string,
  newTest?: string,
  preload?: string
}
export type Action = TestAction | PreloadAction

// Initial Reducer State when none provided
const initialState: State = { test: 'hoi' }

export default (state: State = initialState, action: Action): State => {
  switch (action.type) {
    case preloadActionType: {
      return {
        ...state,
        preload: `pre:${action.value}`
      }
    }
    case `${preloadActionType}_SUCCESS`: {
      return {
        ...state,
        preload: `pre:done:${action.value}`
      }
    }
    case testActionType: {
      return {
        ...state,
        newTest: `sync:${action.value}`
      }
    }
    case `${testActionType}_SUCCESS`: {
      return {
        ...state,
        newTest: action.value
      }
    }
    default: {
      return state
    }
  }
}
