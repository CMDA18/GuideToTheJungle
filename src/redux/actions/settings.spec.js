import { testActionType, createTestAction, createPreloadAction } from './settings'

describe('Settings Actions', () => {
  it('Provides a set of constants', () => {
    expect(testActionType).toBe('TEST')
  })

  it('Provides a set of action creators', () => {
    expect(createTestAction('hoi')).toMatchSnapshot()
    expect(createPreloadAction('preload-hoi')).toMatchSnapshot()
  })
})
