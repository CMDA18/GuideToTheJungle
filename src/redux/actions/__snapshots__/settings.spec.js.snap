// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Actions Provides a set of action creators 1`] = `
Object {
  "type": "TEST",
  "value": "hoi",
}
`;

exports[`Settings Actions Provides a set of action creators 2`] = `
Object {
  "type": "PRELOAD",
  "value": "preload-hoi",
}
`;
